app/routes/transfer/direct/$chainId.$assetId/$address.result.tsx

```diff
@@ -0,0 +1,51 @@
+import { useNavigate } from 'react-router-dom';
+
+import { type ClientLoaderFunction, useLoaderData } from '@remix-run/react';
+import { $params, $path } from 'remix-routes';
+
+import { useGlobalContext } from '@/common/providers';
+import { MainButton } from '@/common/telegram/MainButton';
+import { pickAsset } from '@/common/utils';
+import { Icon, MediumTitle, TitleText } from '@/components';
+
+// Query params for /transfer/direct/:chainId/:assetId/result?amount=__value__
+export type SearchParams = {
+  amount: string;
+};
+
+export const clientLoader = (({ params, request }) => {
+  const url = new URL(request.url);
+  const amount = url.searchParams.get('amount') || '';
+  const data = $params('/transfer/direct/:chainId/:assetId/:address/result', params);
+
+  return { amount, ...data };
+}) satisfies ClientLoaderFunction;
+
+const Page = () => {
+  const { address, amount, assetId, chainId } = useLoaderData<typeof clientLoader>();
+
+  const navigate = useNavigate();
+  const { assets } = useGlobalContext();
+
+  const selectedAsset = pickAsset(chainId, assetId, assets);
```

Comment:
Yeah, I agree. For now asset is determined based on query params.

Reply:
This change was made to prevent imperative `selectedAsset` management, like it was before. With asset being encoded in url, it's simpler to maintain consistent views per screen. IMO telenova will mostly be app with screen-by-screen flows and some event-based management looks over-engineered for me.